sycl/include/sycl/atomic_ref.hpp

```diff
@@ -157,8 +158,16 @@ class atomic_ref_base {
   }
 
 #ifdef __SYCL_DEVICE_ONLY__
+#if defined(__SPIR__)
```

Comment:
Why is feature test macro `SYCL_EXT_ONEAPI_ADDRESS_CAST` not defined for the `__SPIR__` case?

Reply:
Extension implementation seems to have missed that completely. I'm making other changes to the extension, will work on that outside this PR.